examples/pip_parse_vendored/requirements.bzl

```diff
@@ -5,10 +5,12 @@ from //:requirements.txt
 """
 
 load("@python39//:defs.bzl", "interpreter")
-load("@rules_python//python/pip_install:pip_repository.bzl", "whl_library")
+load("@rules_python//python/pip_install:pip_repository.bzl", "group_library", "whl_library")
 
 all_requirements = ["@pip_certifi//:pkg", "@pip_charset_normalizer//:pkg", "@pip_idna//:pkg", "@pip_requests//:pkg", "@pip_urllib3//:pkg"]
 
+all_requirement_groups = {}
```

Comment:
Ideally we would call `pip install report --dry-run` during creating `requirements.bzl` file and detect these `groups` programatically. What do you think?

Reply:
I'd love that honestly. In my ideal world this would Just Work (TM) magically without users having to think about cycles. If we can run that command maybe during the initial `pip_parse` invocation we could definitely try to drive groups automatically.